### PR TITLE
Sync with fe86ed4

### PR DESCRIPTION
**Pull request by qnzhou**

### CODE REVIEW

**Review by jdumas — APPROVED** [review]

File: modules/core/src/mesh_cleanup/split_edges.cpp [hunk 1]
```diff
@@ -108,12 +108,12 @@ std::vector<Index> split_edges(
     std::vector<std::array<Index, 2>> parent_edge(
         num_output_vertices,
         {invalid<Index>(), invalid<Index>()});
-    tbb::parallel_for(Index(0), num_edges, [&](Index eid) {
+    for (Index eid = 0; eid < num_edges; eid++) {
```
Comment (jdumas) [inline modules/core/src/mesh_cleanup/split_edges.cpp:111]: I don't understand why this change is necessary?

Comment (qnzhou) [reply modules/core/src/mesh_cleanup/split_edges.cpp:111]: Thread sanitizer is complaining about race condition. But yes, I think it is a false alarm.

Comment (jdumas) [reply modules/core/src/mesh_cleanup/split_edges.cpp:111]: It's actually not a false alarm. Looking more closely at the code you are iterating over each edges, writing the vertex->edge relationship in the `parent_edge` array. When multiple edges are incident to the same vertex this parent will be overwritten multiple time, with different edge indices each time.

Comment (qnzhou) [reply modules/core/src/mesh_cleanup/split_edges.cpp:111]: Actually, I think you are right. For `split_long_edges`, every edge split point has a unique parent edge, so there is no race condition. However, for `remove_degenerate_facets`, an edge split points may belong to multiple collinear edges, which leads to the race condition. Let me fix that.
